Add org permissions fetcher (#53)
ComplianceAsCode/auditree-arboretum: arboretum/permissions/README.md

## Fetchers

> ### Organization Integrity (Repository Collaborators and Forks)

Reviewer: ```suggestion
### Organization Integrity Permissions
```

* Class: [GithubOrgPermissionFetcher][fetch-org-permissions]
* Purpose: Writes the details of collaborators and repository forks in Github organizations to the evidence locker. This fetcher class is only meant for use with Github or Github Enterprise organizations.
* Behavior: For each Github organization specified an evidence file per collaborator type (affiliation) and an evidence file with the list of repository forks are stored in

> the locker. The default is to retrieve all collaborators by affiliation and all forks from all repositories in each specified Github organization.  TTL is set to 1 day.

Reviewer: ```suggestion
* Behavior: For each Github organization specified, Github collaborator and Github fork evidence files per collaborator type (affiliation) are stored in
the locker containing details for the specified repositories in the organization. The default is to
retrieve all collaborators and all forks by affiliation from all repositories in each specified Github organization.  TTL is set to 1 day.
```

           * Optional
           * List of strings in the form of `["my-repo", "my-other-repo"]`.
           * Defaults to all repositories in the organization.

>            * Use if looking to filter collaborator and fork evidence to a subset of repositories in the organization otherwise do not include.

Reviewer: ```suggestion
           * Use if looking to filter permissions evidence to a subset of repositories in the organization otherwise do not include.
```

  from arboretum.permissions.fetchers.github.fetch_org_collaborators import GithubOrgCollaboratorsFetcher
  ```

### Organization Integrity (Repository Forks)

> * Class: [GithubRepoForksFetcher][gh-org-fetcher]

Reviewer: This isn't the `gh-org-fetcher`.  You should provide a proper link and reference it here correctly.

Also, it seems like it would make sense to rename the class as GithubOrgForksFetcher?  Suggest renaming the class and changing the link ref for `gh-org-fetcher` to `gh-org-collaborators-fetcher` and adding a `gh-org-forks-fetcher` link reference to button it all up.

* Purpose: Writes the repository forks in Github organizations to the evidence locker. This fetcher class is only meant for use with Github or Github Enterprise organizations.

> * Behavior: For each Github organization specified, an evidence file  is stored in

Reviewer: ```suggestion
* Behavior: For each Github organization specified, an evidence file is stored in
```

the locker containing the forks for the specified repositories in the organization. The default is to
retrieve all forks from all repositories in each specified Github organization.  TTL is set to 1
day.
* Configuration elements:
  * `org.permissions.org_integrity.orgs`
     * Required
     * List of dictionaries each containing Github organization retrieval configuration.
        * `url`
           * Required
           * String in the form of `"https://github.com/my-org"` or `"https://github.<company>.com/my-org"`.
           * Use to define the Github organization url to use.
        * `repos`
           * Optional
           * List of strings in the form of `["my-repo", "my-other-repo"]`.
           * Defaults to all repositories in the organization.

>            * Use if looking to filter collaborator evidence to a subset of repositories in the organization otherwise do not include.

Reviewer: By doing this you're tying the collaborators fetcher and the forks fetcher together.  If someone chooses to run both of these fetchers in their environment then they're forced to run them for the exact same set of orgs and repos.  Is that the desired behavior?

[auditree-framework]: https://github.com/ComplianceAsCode/auditree-framework
[auditree-framework documentation]: https://complianceascode.github.io/auditree-framework/
[usage]: https://github.com/ComplianceAsCode/auditree-arboretum#usage
[fetch-org-permissions]: https://github.com/ComplianceAsCode/auditree-arboretum/blob/main/arboretum/permissions/fetchers/github/fetch_org_permissions.py

> [repository-permissions]: https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization[org-collaborators-check]: https://github.com/ComplianceAsCode/auditree-arboretum/blob/main/arboretum/permissions/checks/test_org_collaborators.py

Reviewer: `repository-permissions` doesn't seem right here.  I think you're breaking the link.
